cjdns-core/src/encoding.rs

```diff
@@ -234,28 +232,30 @@ fn read_n_bits_from_position_into_u32(vecc: &Vec<u8>, position: u32, bits_amount
 ///
 /// Accepts bytes array, parses it and returns vector of `EncodingSchemeForm`s.
 pub fn deserialize_forms(vecbytes: &Vec<u8>) -> Result<Vec<EncodingSchemeForm>, Error> {
```

Comment:
@a-kordys better `vecbytes: &[u8]`, isn't it?

Reply:
Definitely. To me, `&Vec` is a quite strange type for a public API. If you need ownership - it should be just `Vec` (not this case), if not - `&[]` is sufficient. Form `&Vec` requires caller to have specifically `Vec` but it's capabilitiesa are not used at all because of that shared reference.